examples/src/js/auto-styled/index.js

```diff
@@ -0,0 +1,18 @@
+import Component from '@biotope/element';
+import * as styles from './styles.scss';
+
+class ExampleButton extends Component {
+  constructor() {
+    super();
+    this.styleContent = styles;
```

Comment:
What would be better, the approach above, or this one?
```javascript
this.setStyle(styles);
```
@SheepFromHeaven @timomayer

Reply:
i vote for this.setStyle(styles);

Reply:
I also prefer `setStyle` but I'm having trouble with the following:

Pros for `setStyle`:
- feels cleaner since it does not just set the value - it also triggers a render
- it's consistent with `setState` (which does something very similar - changes a variable and triggers a new render)

Cons for `setStyle`:
- in typescript, it forces the developer to add a `constructor` (or some other function) to use it, instead of just a simple assignment:
```typescript
class MyButton extends Component {
  protected styleContent = styles;
}
```
- variable assignments are more commonly done outside the `render` function, so the chances of infinite-loops are lower (because using `setStyle` inside the render will trigger an infinite loop)

Reply:
@timomayer @SheepFromHeaven bump

Reply:
@tiagomapmarques I still would vote for the function.
for your **first** con: I see this problem but for me it feels acceptable
for your **second** con: I don't see the difference to setState which we also chose.

Reply:
@tiagomapmarques now that I think about it, the non functional approach is much better for extending elements. So I would call the property `styles` and vote for it!

Reply:
@timomayer can we agree on that?

Reply:
i can agree with using the variable and not function especially the second con is valid in my opinion